src/Atmos/Model/AtmosModel.jl

```diff
@@ -244,23 +244,34 @@ struct DYCOMS_BC <: BoundaryCondition
 end
 function boundarycondition!(bl::AtmosModel{T,M,R,S,BC,IS}, stateP::Vars, diffP::Vars, auxP::Vars,
     nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, state1::Vars, diff1::Vars, aux1::Vars) where {T,M,R,S,BC <: DYCOMS_BC,IS}
+    # stateM is the 𝐘⁻ state while stateP is the 𝐘⁺ state at an interface. 
+    # at the boundaries the ⁻, minus side states are the interior values
+    # state1 is 𝐘 at the first interior nodes relative to the bottom wall 
+    
+    # Get values from minus-side state
     ρM = stateM.ρ 
     UM, VM, WM = stateM.ρu
     EM = stateM.ρe
     QTM = stateM.moisture.ρq_tot
     uM, vM, wM  = UM/ρM, VM/ρM, WM/ρM
     q_totM = QTM/ρM
     UnM = nM[1] * UM + nM[2] * VM + nM[3] * WM
+    
+    # Assign reflection wall boundaries (top wall)
     stateP.ρu = SVector(UM - 2 * nM[1] * UnM, 
```

Comment:
Is it possible to hide the way the no-flux (and b.c. in general) from the person imposing the boundary conditions?
A user who sets up a problem should only need to worry about the type of b.c. and not how it is implemented.

For example, hide this implementation behind something like stateP.\rhou = bc(NO_FLUX)

Reply:
@smarras79 There is a separate PR https://github.com/climate-machine/CLIMA/pull/412 that moves towards this